src/components/fidelity_bond/FidelityBondDetailsSetupForm.tsx

```diff
@@ -0,0 +1,582 @@
+
+import { WalletInfo, CurrentWallet, Utxos, Account } from '../../context/WalletContext'
+// @ts-ignore
+import { useSettings } from '../../context/SettingsContext'
+
+// @ts-ignore
+import Balance from '../../components/Balance'
+// @ts-ignore
+import ToggleSwitch from '../../components/ToggleSwitch'
+import { useBalanceSummary, WalletBalanceSummary } from '../../hooks/BalanceSummary'
+
+import Sprite from './../Sprite'
+import UtxoSelector from './UtxoSelector'
+import AccountSelector from './AccountSelector'
+import LockdateForm, { toYearsRange, lockdateToTimestamp, DEFAULT_MAX_TIMELOCK_YEARS } from './LockdateForm'
+
+import { routes } from '../../constants/routes'
+import * as Api from '../../libs/JmWalletApi'
+
+const timeUtils = (() => {
+  type Milliseconds = number
+  type UnitKey = 'year' | 'month' | 'day' | 'hour' | 'minute' | 'second'
+
+  type Units = {
+    [key in UnitKey]: Milliseconds
+  }
+
+  const units: Units = {
+    year: 24 * 60 * 60 * 1_000 * 365,
+    month: (24 * 60 * 60 * 1_000 * 365) / 12,
+    day: 24 * 60 * 60 * 1_000,
+    hour: 60 * 60 * 1_000,
+    minute: 60 * 1_000,
+    second: 1_000,
+  }
+
+  type Locales = Intl.UnicodeBCP47LocaleIdentifier | Intl.UnicodeBCP47LocaleIdentifier[]
+  const timeElapsed = (d1: Milliseconds, d2: Milliseconds = Date.now(), locales: Locales = 'en') => {
+    const rtf = new Intl.RelativeTimeFormat(locales, { numeric: 'auto' })
+    const elapsedInMillis: Milliseconds = d1 - d2
+
+    for (let k of Object.keys(units) as UnitKey[]) {
+      const limit: number = units[k]
+      if (Math.abs(elapsedInMillis) > limit) {
+        return rtf.format(Math.round(elapsedInMillis / limit), k)
+      }
+    }
+
+    return rtf.format(Math.round(elapsedInMillis / units['second']), 'second')
+  }
+
+  return {
+    timeElapsed,
+  }
+})()
+
+interface SelectAccountStepProps {
+  walletInfo: WalletInfo
+  onSelected: (account: Account | null) => void
+}
+
+const SelectAccountStep = ({ walletInfo, onSelected }: SelectAccountStepProps) => {
+  const accounts = useMemo(() => walletInfo.data.display.walletinfo.accounts, [walletInfo])
+  const utxos = useMemo(() => walletInfo.data.utxos.utxos, [walletInfo])
+
+  // TODO: this is a common pattern - try to generalize
+  const utxosByAccount = useMemo(() => {
+    return utxos.reduce((acc, utxo) => {
+      const key = `${utxo.mixdepth}`
+      acc[key] = acc[key] || []
+      acc[key].push(utxo)
+      return acc
+    }, {} as { [key: string]: Utxos })
+  }, [utxos])
+
+  const availableAccounts = useMemo(() => {
+    const accountsWithUtxos = Object.keys(utxosByAccount)
+    return accounts.filter((it) => accountsWithUtxos.includes(it.account))
+  }, [utxosByAccount, accounts])
+
+  const selectableAccounts = useMemo(() => {
+    return accounts.map((it) => ({
+      ...it,
+      disabled: !availableAccounts.includes(it),
+      utxos: utxosByAccount[it.account] || [],
+    }))
+  }, [accounts, availableAccounts, utxosByAccount])
```

Comment:
Is there ever an account that's available but not selectable? If not, I'd say let's not differentiate those two concepts and just have a single list of accounts including all info necessary.

Reply:
Yes. You should have experienced this by yourself in regtest: At first, only Jar Zero contains utxos and it is the only one "selectable".
What is your suggestion: Should only Jar Zero be shown? Or should all be "selectable" and "No utxos" phrase shown in the next step?